### PR TITLE
Fix onScroll event type

### PR DESCRIPTION
**Pull request by Hazeman27**
# Summary

Fixes type of the event of the `onScroll` function.

Issue: #1627

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅     |
| Android |    ✅     |

## Checklist

- [X] I have tested this on a device and a simulator
- [X] I added the documentation in `README.md`
- [ ] I mentioned this change in `CHANGELOG.md` **(Didn't find `CHANGELOG.md` in the repo)**
- [X] I updated the typed files (TS and Flow)
- [X] I added a sample use of the API in the example project (`example/App.js`) **(Example already existed)**

### CODE REVIEW

File: docs/Reference.md [hunk 1]
```diff
@@ -577,6 +578,31 @@ url
...
+  source={{ uri: 'https://reactnative.dev' }}
+  onScroll={syntheticEvent => {
+    const { contentOffset } = syntheticEvent.nativeEvent
+    console.table(contentOffset)
+  }}
+/>
+```
+
+Function will receive event object of `NativeSyntheticEvent<NativeScrollEvent>`
+type as an argument.
```
Comment (TheAlmightyBob) [inline docs/Reference.md:602]: The documentation for other events also lists the properties of the native event. Could you add that?

**Review by TheAlmightyBob — CHANGES_REQUESTED** [review]
Looks good, just one small comment on the docs (thanks for adding documentation though!)

**Review by TheAlmightyBob — APPROVED** [review]

**Comment by Hazeman27** [comment]
Still waiting for review and merge 😢

**Comment by TheAlmightyBob** [comment]
@jamonholmgren @Titozzz This seems like a straightforward types/documentation fix :)

**Review by Titozzz — APPROVED** [review]

**Comment by Titozzz** [comment]
> Still waiting for review and merge 😢

Sorry 😭

**Pull request merged**